components/speedreader/rust/lib/src/readability/src/dom.rs

```diff
@@ -190,62 +201,68 @@ pub fn has_nodes(handle: &Handle, tag_names: &[&'static LocalName]) -> bool {
     false
 }
 
+#[inline]
+pub fn get_single_element_by_tag(handle: &Handle, tag: &LocalName) -> Option<Handle> {
+    let mut elems = handle
+        .children()
+        .filter(|child| child.as_element().is_some());
+    if elems.clone().count() == 1 {
```

Comment:
Agreed. So the point of this part is sites like mobile.slate.com wrap every paragraph element in a div, and this can confuse the scoring algorithm. So we hoist those paragraph elements up and remove the divs. So I think the most clear thing we can do is rename the function as you suggested, and move that elemet count == 1 check into the scorer.rs, since it has more to do with the algorithm.